### PR TITLE
rmi by image id

### LINKED ISSUE #157: rmi by image id
## Issue Description

Type: *feature request*
currently, we support delete image by name only, which provides ability of untagging when there are many images with same id. Thus we can't delete image just one step in some situations. I think provide a new feature for deleting image by image id is reasonable. 
### Describe what feature you want
```
[root@jiangnan001 ~]# sealer images
+----------------------------------+----------------------------------------------------------------------+
|             IMAGE ID             |                              IMAGE NAME                              |
+----------------------------------+----------------------------------------------------------------------+
| 8233ad647e35d6ec32d98efe45467cb5 | registry.cn-hongkong.aliyuncs.com/sealyun/kubernetes-amd64-1:v1.19.9 |
| 8233ad647e35d6ec32d98efe45467cb5 | registry.cn-hongkong.aliyuncs.com/sealyun/kubernetes-amd64:v1.19.9   |
```
sealer rmi 8233ad647e35d6ec32d98efe45467cb5 (failed, there are more than one image)
sealer rmi 8233ad647e35d6ec32d98efe45467cb5 -f (success)

### PR DESCRIPTION
**Pull request by chaunceyjiang**
### Describe what this PR does / why we need it

rmi by image id

### Does this pull request fix one issue?

Fixes #157

### CODE REVIEW

File: sealer/cmd/rmi.go [hunk 1]
```diff
@@ -15,28 +15,52 @@
...
+	force bool
+}
+
+var opts removeImageFlag
+
 // rmiCmd represents the rmi command
 var rmiCmd = &cobra.Command{
 	Use:     "rmi",
-	Short:   "rmi delete local image",
+	Short:   "Remove one or more images",
```
Comment (fanux) [inline sealer/cmd/rmi.go:36]: Remove local images by name or ID

Comment (chaunceyjiang) [reply sealer/cmd/rmi.go:36]: > Remove local images by name or ID

ok

File: image/default_image.go [hunk 2]
```diff
@@ -246,7 +260,7 @@ func (d DefaultImageService) Delete(imageName string) error {
```
Comment (touchuyht) [inline image/default_image.go:260]: The if-statement `if imageTagCount != 1` should be changed into `if imageTagCount != 1 && !d,ForceDeleteImage` for in the force delete image scene, we do not want to terminate image deletion here.

File: image/default_image.go [hunk 3]
```diff
@@ -246,7 +260,7 @@ func (d DefaultImageService) Delete(imageName string) error {
 
 	for _, layer := range image.Spec.Layers {
 		layerID := store.LayerID(layer.Hash)
-		if isLayerDeletable(layer2ImageNames, layerID) {
+		if isLayerDeletable(layer2ImageNames, layerID) || d.ForceDeleteImage {
```
[the remark this answers is not in the document]
Comment (chaunceyjiang) [reply image/default_image.go:263]: Do we need to implement this function?

[the remark this answers is not in the document]
Comment (touchuyht) [reply image/default_image.go:263]: At present, sealer images do not have that complex dependent relationships when compared to docker images. So I think we do not need to implement this function.

[the remark this answers is not in the document]
Comment (chaunceyjiang) [reply image/default_image.go]: ok

**Review by touchuyht — APPROVED** [review]

File: image/default_image.go [hunk 4]
```diff
@@ -197,38 +200,49 @@ func (d DefaultImageService) Delete(imageName string) error {
...
-	//1.untag image
-	err = imageutils.DeleteImage(imageName)
-	if err != nil {
-		return fmt.Errorf("failed to untag image %s, err: %s", imageName, err)
+	if strings.Contains(imageName, ":") {
+		//1.untag image
+		if err = imageutils.DeleteImage(imageName); err != nil {
+			return fmt.Errorf("failed to untag image %s, err: %w", imageName, err)
+		}
+		image, err = imageutils.GetImageByID(imageMetadata.ID)
```
Comment (fanux) [inline image/default_image.go:216]: We should not ignore any errors

Comment (chaunceyjiang) [reply image/default_image.go:216]: > We should not ignore any errors
yes ,i,know ,but I should not have  ignore any errors.

**Pull request merged**